cross-modal-search/README.md

```diff
@@ -70,128 +58,146 @@ for encoding image and text respectively.
-
-Internally, `TextEncoder` targets `ImageVectorIndexer` and `ImageEncoder` targets `TextVectorIndexer`.
-`ImageVectorIndexer` and `TextVectorIndexer` map to a common Embedding Space. (To Jina it means having common dimensionality).
+### 🔎 Step 3: Query your data
+Jina offers several ways to search (query) your data. In this example, we show two of the most common ones. In a production environment, you would only choose one which suits your use case best. 
 
-### With REST API
+#### Using a REST API
+Begin by running the following command to open the REST API interface.
 
 ```sh
 python app.py -t query_restful
 ```
 
-Then:
+You should open another terminal window and paste the following command. 
 
 ```sh
-curl --request POST -d '{"top_k": 10, "mode": "search",  "data": ["hello world"]}' -H 'Content-Type: application/json' 'http://0.0.0.0:45678/search'
-````
+curl --request POST -d '{"top_k": 5, "mode": "search",  "data": ["hello world"]}' -H 'Content-Type: application/json' 'http://0.0.0.0:45678/search'
+```
 
-Or use [Jinabox](https://jina.ai/jinabox.js/) with endpoint `http://127.0.0.1:45678/search`
+Once you run this command, you should see a JSON output returned to you. This contains the five most semantically similar images sentences to the text input you provided in the `data` parameter.
+Note, that the toy-data only contains two images.
+Feel free to alter the text in the 'data' parameter and play around with other queries (this is only fun with the large dataset)! For a better understanding of the parameters see the table below. 
+|  |  |
+|--|--|
+| `top_k` | Integer determining the number of sentences to return |
+| `mode` | Mode to trigger in the call. See [here](https://docs.jina.ai/chapters/rest/) for more details |
+| `data` | Text input to query |
+ 
+#### Using Jina Box; our frontend search interface
 
+**Jina Box** is a light-weight, highly customizable JavaScript based front-end search interface. To use it for this example, begin by opening the REST API interface. 
 
-## Use Docker image from the jina hub
+```sh
+python app.py -t query_restful
+```
 
+In your browser, open up the hosted Jina Box on [jina.ai/jinabox.js](https://jina.ai/jinabox.js/). In the configuration bar on the left-hand side, choose a custom endpoint and enter the following: `http://127.0.0.1:45678/search`.
+You can type search queries into the text box on the right-hand side!
+
+______
+
+## 🌀 Flow diagram
+This diagram provides a visual representation of the Flows in this example; Showing which executors are used in which order.
+Remember, our goal is to compare vectors representing the semantics of images with vectors encoding the semantics of short text descriptions.
+
+### Indexing
+![](visualizations/index-flow.png)
+As you can see, the Flow that indexes the data contains three parallel branches. 
+The upper branch contains only a key-value indexer for the images that we use as a lookup (like a Dictionary in Python).
```

Comment:
Could you maybe have 3 bullet points one for each branch? I know what you meant, I feel it's a bit dense paragraph.

Reply:
5f33ee0ac0a037bf57c8c44113a7548b79a90bf5

Reply:
Sorry its this one https://github.com/jina-ai/examples/pull/614/commits/0b84f32275def95b4d63334949e2e044abd4f315